### PR TITLE
chore: Add missing javadocs in Smart Contract Service

### LINKED ISSUE #15347: Add missing Javadocs in Smart Contract Service

### PR DESCRIPTION
**Pull request by petreze**
**Description**:
This PR adds missing javadocs to Smart Contract Service. This won't close the issue as there were quite many of them (more than 1k missing javadocs) and they will be split in several PRs
Those are found by going to `Analyze -> Inspect code`

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by andrewb1269hg — APPROVED** [review]

**Review by andrewb1269hg — APPROVED** [review]
Review covers:
- `hedera-node/hedera-smart-contract-service-impl/src/main/java/module-info.java`
- `hedera-node/hedera-smart-contract-service/src/main/java/module-info.java`
Looks good.

**Comment by andrewb1269hg** [comment]
Please review the Codacy findings.

**Review by andrewb1269hg — DISMISSED** [review]
Reviewed the following files:
`hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/ContractServiceImpl.java`
`hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/hevm/HederaWorldUpdater.java`

**Review by andrewb1269hg — APPROVED** [review]
Reviewed following files:
`hedera-node/hedera-smart-contract-service-impl/src/main/java/module-info.java`
`hedera-node/hedera-smart-contract-service/src/main/java/module-info.java`

These two files look good.

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/gas/DispatchType.java [hunk 1]
```diff
@@ -30,6 +30,7 @@
 /**
  * Enumerates the types of child transactions that can be dispatched to the HTS system contract.
  */
+@SuppressWarnings("MissingJavadoc")
```
Comment (mhess-swl) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/gas/DispatchType.java:33]: Odd, why is this necessary?

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/operations/CustomizedOpcodes.java [hunk 2]
```diff
@@ -24,6 +24,7 @@
  * <p>Please see <a href="https://ethereum.org/en/developers/docs/evm/opcodes/">here</a>
  * for a review of the EVM opcodes.
  */
+@SuppressWarnings("ALL")
```
Comment (mhess-swl) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/operations/CustomizedOpcodes.java:27]: Do we really want to suppress _all_ warnings here?

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/operations/DelegatingOperation.java [hunk 3]
```diff
@@ -28,6 +28,9 @@
 public class DelegatingOperation implements Operation {
     private final Operation delegate;
 
+    /**
+     * @param delegate the delegate operation
```
Comment (mhess-swl) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/operations/DelegatingOperation.java:32]: Nit:

```suggestion
     * @param delegate the given operation
```

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/transfer/CallStatusStandardizer.java [hunk 4]
```diff
@@ -37,12 +37,21 @@
  * Standardizes the failure status of a call to a HTS transfer system contract.
  */
 public class CallStatusStandardizer {
+    /**
+     *
```
Comment (mhess-swl) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/transfer/CallStatusStandardizer.java:41]: empty?

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractHandlers.java [hunk 5]
```diff
@@ -22,6 +22,9 @@
 import javax.inject.Inject;
 import javax.inject.Singleton;
 
+/**
+ *
```
Comment (mhess-swl) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractHandlers.java:26]: Empty?

**Review by mhess-swl — APPROVED** [review]
Just a few very minor comments, but overall LGTM

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/operations/DelegatingOperation.java (hunk 3 above)
Comment (petreze) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/operations/DelegatingOperation.java:32]: Applied

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractHandlers.java (hunk 5 above)
Comment (petreze) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractHandlers.java:26]: Added

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/transfer/CallStatusStandardizer.java (hunk 4 above)
Comment (petreze) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/transfer/CallStatusStandardizer.java:41]: Added

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/gas/DispatchType.java (hunk 1 above)
Comment (petreze) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/gas/DispatchType.java:33]: I added this because I thought the variables are self-explanatory and did not need javadocs but I will add some useful comments

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/operations/CustomizedOpcodes.java (hunk 2 above)
Comment (petreze) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/operations/CustomizedOpcodes.java:27]: added comments explaining the opcodes

**Review by mhess-swl — APPROVED** [review]
LGTM! ✅

**Review by andrewb1269hg — APPROVED** [review]
Reviewed 2 files that only added header comments:
`hedera-node/hedera-smart-contract-service-impl/src/main/java/module-info.java`
`hedera-node/hedera-smart-contract-service/src/main/java/module-info.java`

**Review by mhess-swl — DISMISSED** [review]

**Review by andrewb1269hg — DISMISSED** [review]
Reapproved.

**Review by andrewb1269hg — APPROVED** [review]
Reapproved.

**Review by mhess-swl — DISMISSED** [review]

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/ContextTransactionProcessor.java [hunk 6]
```diff
@@ -69,10 +69,24 @@ public class ContextTransactionProcessor implements Callable<CallOutcome> {
...
+     * @param contractsConfig the contracts configuration to use
+     * @param configuration the configuration to use
+     * @param hederaEvmContext the hedera EVM context
+     * @param addOnTracers all operation tracer callbacks
+     * @param evmActionTracer the EVM action tracer
+     * @param worldUpdater the world updater for the transaction
+     * @param hevmTransactionFactory the factory for EVM transaction
+     * @param feesOnlyUpdater if base commit fails, a fees-only updater
+     * @param processor a map from the version of the Hedera EVM to the transaction processor
+     * @param customGasCharging the Hedera gas charging logic
```
Comment (lukelee-sl) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/ContextTransactionProcessor.java:88]: no return value here

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/ContextTransactionProcessor.java [hunk 7]
```diff
@@ -69,10 +69,24 @@ public class ContextTransactionProcessor implements Callable<CallOutcome> {
     private final TransactionProcessor processor;
     private final EvmActionTracer evmActionTracer;
     private final RootProxyWorldUpdater rootProxyWorldUpdater;
-    public final HevmTransactionFactory hevmTransactionFactory;
+    private final HevmTransactionFactory hevmTransactionFactory;
```
Comment (lukelee-sl) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/ContextTransactionProcessor.java:72]: it seems this pr does have at least one change that is not changing a comment?

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/scope/HederaOperations.java [hunk 8]
```diff
@@ -38,6 +38,9 @@
  * Provides the Hedera operations that only a {@link ProxyWorldUpdater} needs (but not a {@link DispatchingEvmFrameState}.
  */
 public interface HederaOperations {
+    /**
+     * Default contract id to return when the searched one is not found,
```
Comment (lukelee-sl) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/scope/HederaOperations.java:42]: this comment seems incorrect compared to how this constant is actually used.

**Review by lukelee-sl — DISMISSED** [review]
LGTM.  hard to do a through job of reviewing but I went through as much as possible

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/ContextTransactionProcessor.java (hunk 7 above)
Comment (petreze) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/ContextTransactionProcessor.java:72]: I must've forgotten this, I guess I just saw a possible minor change to make.

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/scope/HederaOperations.java (hunk 8 above)
Comment (petreze) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/scope/HederaOperations.java:42]: Does something like "A contract id to indicate that a given ContractId has a mismatch with the config" feels more appropriate? @lukelee-sl

Comment (petreze) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/scope/HederaOperations.java:42]: Changed to _A contract id to indicate that a given ContractId has a mismatch with the config_

**Review by lukelee-sl — APPROVED** [review]
LGTM

**Review by mhess-swl — APPROVED** [review]
